Documentation/internals/hooks.rst

```diff
@@ -0,0 +1,16 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+eBPF Hooks
+==========
+
+Cilium uses the following eBPF hooks to attach eBPF programs to the kernel:
+
+- ``BPF_PROG_TYPE_XDP``
+- ``BPF_PROG_TYPE_SCHED_ACT``
+- ``BPF_PROG_TYPE_CGROUP_SOCK_ADDR``
+- ``BPF_PROG_TYPE_SOCK_OPS``
+- ``BPF_PROG_TYPE_SK_MSG``
```

Comment:
What do we mean by "hooks"? I would call these program types, some of which can attach at multiple locations in the kernel, take `BPF_PROG_TYPE_CGROUP_SOCK_ADDR` for example which we attach at `connect4`, `bind4`, ect.

Reply:
Makes sense yeah. I'll update to define this as the program types. I don't plan on documenting all of the individual function attachment locations in this PR.